File: algorithms/198. House Robber/solution.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        p1 = p2 = 0
        for n in nums:
            p1, p2 = p2, max(p1 + n, p2)
        return p2

File: algorithms/198. House Robber/solution.cc
class Solution {
public:
    int rob(vector<int>& nums) {
        int p1 = 0, p2 = 0, tmp;
        for ( int n : nums ) {
            tmp = p2;
            p2 = max(p1 + n, p2);
            p1 = tmp;
        }
        return p2;
    }
};
